### PR TITLE
feat(approval): support search query and filter by resource type and account_type

### LINKED ISSUE #52: Support search query and more filters in list approvals API
**Summary**
Similar to the problem mentioned in https://github.com/goto/guardian/issues/5, some users can receive many approvals, which would be too heavy to be returned in one API call. Although pagination has helped, we also need to support search query and some filters. 

**Proposed solution**
Additionally to existing filters, these new filters can be added
- search `?q=foobar`
   - lookup to `appeal.account_id`, `appeal.role`, and `resource.urn`
- filter by resource types `?resource_types=dataset&resource_types=table`
- filter by account types `?account_types=user&account_types=serviceAccount`

Note: Performance shouldn’t be impacted

### PR DESCRIPTION
**Pull request by rahmatrhd**
Proton PRs: https://github.com/goto/proton/pull/28 and https://github.com/goto/proton/pull/30

closes #52 

## Changes:
* support search query for list approvals
* add more filters
  * `resource_types`
  * `account_types`
* add `total` count in list approvals response

### CODE REVIEW

File: internal/store/postgres/approval_repository.go [hunk 1]
```diff
@@ -37,19 +37,32 @@ func (r *ApprovalRepository) ListApprovals(ctx context.Context, conditions *doma
...
 		db = db.Where(`"approvers"."email" = ?`, conditions.CreatedBy)
 	}
 	if conditions.Statuses != nil {
 		db = db.Where(`"approvals"."status" IN ?`, conditions.Statuses)
 	}
+	if conditions.Q != "" {
+		db = db.Where(db.
+			Or(`"Appeal"."account_id" LIKE ?`, fmt.Sprintf("%%%s%%", conditions.Q)).
+			Or(`"Appeal"."role" LIKE ?`, fmt.Sprintf("%%%s%%", conditions.Q)).
+			Or(`"Appeal__Resource"."urn" LIKE ?`, fmt.Sprintf("%%%s%%", conditions.Q)),
```
Comment (bsushmith) [inline internal/store/postgres/approval_repository.go:54]: Is `Q` being used for generic string input for search kind of feature?  
basically search string in any of these three fields... ?

Comment (rahmatrhd) [reply internal/store/postgres/approval_repository.go:54]: yes, please check https://github.com/goto/guardian/issues/52

**Review by bsushmith — APPROVED** [review]

**Comment by bsushmith** [comment]
LGTM

**Review by bsushmith — APPROVED** [review]

**Comment by bsushmith** [comment]
@rahmatrhd There are no test package related changes for this fix - [fix: fix gorm query building order](https://github.com/goto/guardian/pull/51/commits/a8b0874bde97f53af392d59a02a9294088c0fab1). Since this is a `fix`, I would assume that it got missed through our automated tests. Can we add some if that's the case.

**Comment by rahmatrhd** [comment]
> @rahmatrhd There are no test package related changes for this fix - [fix: fix gorm query building order](https://github.com/goto/guardian/pull/51/commits/a8b0874bde97f53af392d59a02a9294088c0fab1). Since this is a `fix`, I would assume that it got missed through our automated tests. Can we add some if that's the case.

@bsushmith added the test case. Previously, a condition stated before the grouped conditions for `Q` was getting added into the group unexpectedly. Moving the order for `Q` into the beginning is not a proper fix, but it works for now. Adding another grouped condition in the middle in the future might cause the same issue.

**Review by bsushmith — APPROVED** [review]

**Pull request merged**